Libraries/libbom/Headers/bom/bom.h

```diff
@@ -90,6 +93,9 @@ struct bom_tree_context *
 bom_tree_alloc_empty(struct bom_context *context, const char *variable_name);
 
 struct bom_tree_context *
+bom_tree_alloc_size(struct bom_context *context, const char *variable_name, uint32_t index_count);
```

Comment:
I'm a little worried about this one. The semantics are different than from the BOM as a whole: while for a BOM the `index_count` is a hint, it's a hard requirement for a BOM tree. Since the BOM tree doesn't store a free list, the tree is malformed with empty entries. That isn't obvious from the API.